methods/wasserstein.py

```diff
@@ -0,0 +1,49 @@
+from sklearn.neighbors import NearestNeighbors
+from scipy.stats import wasserstein_distance
+import numpy as np
+def split_wasserstein_data(embeddings, test_size, no_trials, leaf_size):
+
+    '''
+    Finds appropriate test sets from encoded texts a.k.a embeddings through the wasserstein metric
+    as described here: https://aclanthology.org/2021.eacl-main.156v2.pdf.
+
+    Steps are as follows
+        1. Embeddings are projects in a nearest neighbor structure
+        2. A random centroid is sampled, from which nearest neighbors using the
+           wasserstein metric is sampled.
+        3. This is the new test set.
+        4. Repeated for no_trials
+
+
+    Parameters:
+        embeddings (np.array): Embeddings of texts - One example, would be one hot encoding
+        test_size (int): The size of the test set
```

Comment:
this is the size of the desired test set, right? i.e. this size will be smaller than n